src/pdm/cli/commands/init.py

```diff
@@ -90,6 +90,21 @@ def ask(self, question: str, default: str) -> str:
             return default
         return termui.ask(question, default=default)
 
+    def ask_project(self, project: Project) -> str:
+        name = self.ask("Project name", project.root.name)
```

Comment:
Also please don't validate the default value or user may feel confused.
Instead, we sanitize the default name by replacing the illegal characters.